### PR TITLE
[#187570742] As a seller, I can delete a product

### PR DESCRIPTION
**Pull request by vegetason**
#### What does this PR do?

A seller should be able to delete a product in his/her collection

#### Describe your changes

I created route for seller to delete a product from a collection

#### PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[*] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Tests
[ ] Other
```

#### How should this be manually tested?

It can be tested using swagger documentation and postman

#### What are the relevant pivotal tracker stories?
[Pivotal Tracker Link](https://www.pivotaltracker.com/n/projects/2703465/stories/187570742)

#### Screenshots (if is appropriate)

### CODE REVIEW

**Review by mugishaj092 — CHANGES_REQUESTED** [review]
Good to go but Solve conflict first

**Review by SHYAKA-Aime — DISMISSED** [review]
As you solved conflicts, the functionality also looks good 👏

**Review by GabrielDushime — DISMISSED** [review]
Kudos great work now a seller can delete his/her product

**Review by jamesndekezi — DISMISSED** [review]
hello after solved conflicts All checks have passed, well done

**Review by mugishaj092 — CHANGES_REQUESTED** [review]
Good job Try to solve circleCI issue

**Review by mugishaj092 — CHANGES_REQUESTED** [review]
CircleCi is not is failing

**Review by GabrielDushime — DISMISSED** [review]
Great work so that this feature is allowing seller to delete a product

**Review by mutsinziisaac — DISMISSED** [review]

**Review by jamesndekezi — CHANGES_REQUESTED** [review]
Good job, but there some conflicts need to be resolved.
Check on that.

**Review by mutsinziisaac — DISMISSED** [review]
looks good

**Review by SHYAKA-Aime — DISMISSED** [review]
Good job, Now it is working Well

**Review by jamesndekezi — DISMISSED** [review]
Well done, this PR looks fine

**Review by GabrielDushime — DISMISSED** [review]

File: src/controllers/product.controller.ts [hunk 1]
```diff
@@ -181,4 +181,23 @@ export const updateProductAvailability = async (
...
+export const deleteProduct=async (req:Request,res:Response)=>{
+
+  const productId=req.params.product_id;
+  const sellerId=req.user?.id;
+  const product = await ProductService.getProductByid(productId)
+  if(product?.seller_id!==sellerId){
+    return res.status(400).json({
+      message:"You are not the owner of the product"
+    })
+  }
```
Comment (didasmbalanya) [inline src/controllers/product.controller.ts:194]: This is reusable code that can be done with a middleware

File: src/controllers/product.controller.ts [hunk 2]
```diff
@@ -181,4 +181,23 @@ export const updateProductAvailability = async (
...
+  else if(product===null){
+    res.status(400).json({message:"The product does not exist"})
+  }
```
Comment (didasmbalanya) [inline src/controllers/product.controller.ts:197]: This is reusable code that can be done with a middleware

File: src/controllers/product.controller.ts [hunk 3]
```diff
@@ -181,4 +181,23 @@ export const updateProductAvailability = async (
...
+  await ProductService.deleteProductById(productId)
+  return res.status(200).json({
```
Comment (didasmbalanya) [inline src/controllers/product.controller.ts:199]: Use the right status code for a DELETE

File: src/routes/product.route.ts [hunk 4]
```diff
@@ -14,4 +15,7 @@ router.patch(
   protectRoute,restrictTo(userRole.seller),
   updateProductAvailability
 );
+
+router.delete('/deleteProduct/:product_id',protectRoute,restrictTo(userRole.seller),deleteProduct)
```
Comment (didasmbalanya) [inline src/routes/product.route.ts:19]: The middleware can also check that its not just a seller role but the right seller

**Review by didasmbalanya — CHANGES_REQUESTED** [review]

**Review by GabrielDushime — APPROVED** [review]
I can see you worked on the requested changes.
No Seller can delete a product from the collection

**Review by mutsinziisaac — APPROVED** [review]

**Review by SHYAKA-Aime — APPROVED** [review]
Looks good and  working well

**Review by jamesndekezi — COMMENTED** [review]
Well done everything seems to work well

**Pull request merged**